Quicksilver/QuicksilverDmgVersioner.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/local/autopkg/python
+#
+# Copyright 2010 Per Olofsson
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""See docstring for AppDmgVersioner class"""
+
+import glob
+import os.path
+
+# PyLint cannot properly find names inside Cocoa libraries, so issues bogus
+# No name 'Foo' in module 'Bar' warnings. Disable them.
+# pylint: disable=E0611
+from Foundation import NSData
+from Foundation import NSPropertyListSerialization
+from Foundation import NSPropertyListMutableContainers
+from Foundation import NSPropertyListXMLFormat_v1_0
+
+# pylint: enable=E0611
+
+# Disable PyLint complaining about 'invalid' camelCase names
+# pylint: disable=C0103
+
+from autopkglib import ProcessorError  # pylint: disable=import-error
+from autopkglib.DmgMounter import DmgMounter  # pylint: disable=import-error
+
+__all__ = ["QuicksilverDmgVersioner"]
+
+
+class QuicksilverDmgVersioner(DmgMounter):
```

Comment:
Is there anything Quicksilver-specific about this processor? Should it be called something more generic, in case it becomes useful for other apps?

I understand this is hopefully a temporarily workaround while we consider core AutoPkg plist-reading improvements, but naming is still worth considering.

Reply:
No, there is nothing specific about them to Quicksilver, they are simply drop-in replacements for `AppDmgVersioner` and `AppPkgCreator`. I offered to do a PR to autopkjg/autopkg for those processors, but Greg and Nick thought that if they were core processors, the "Foundation" parts could be abstracted rather than repeated in each processor, which of course makes sense. As I'm sure you know, you cannot abstract third party processors so I couldn't do that here. 

The reason for this PR is purely to provide a working Quicksilver recipe while that debate continues and hopefully is resolved. Greg showed some interest in doing it.

Reply:
We could, however, call it `FoundationAppDmgVersioner` or something like that if you preferred?

Reply:
Yes, let's go with `FoundationAppDmgVersioner` and `FoundationPkgCreator` just in case other apps leverage the custom processors in the near future. Thank you!

Reply:
OK that's done. It begs the question as to whether these processors should stay in the Quicksilver folder or go into a CommonProcessors folder. I didn't see that you already have one of these. The downside of that for the Quicksilver recipe is that we would need to put a longer path to the processor (e.g. `com.github.homebysix.commonprocessors/FoundationAppDmgVersioner`).

I still hope this is only needed temporarily, so maybe it's not worth it. I'll let you decide :)